subsys/nfc/rpc/client/nfc_rpc_t2t_client.c

```diff
@@ -0,0 +1,169 @@
+
+#include <nrf_rpc/nrf_rpc_cbkproxy.h>
+#include <nrf_rpc/nrf_rpc_serialize.h>
+
+#include <nfc_rpc_ids.h>
+#include <nfc_rpc_common.h>
+
+#include <nfc_t2t_lib.h>
+
+static void nfc_rpc_t2t_cb_rpc_handler(const struct nrf_rpc_group *group,
+				       struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	nfc_t2t_event_t event;
+	const uint8_t *data;
+	size_t data_length;
+	void *context;
+	nfc_t2t_callback_t callback;
+
+	event = nrf_rpc_decode_uint(ctx);
+	data = nrf_rpc_decode_buffer_ptr_and_size(ctx, &data_length);
+	context = (void *)nrf_rpc_decode_uint(ctx);
+	callback = (nfc_t2t_callback_t)nrf_rpc_decode_callback_call(ctx);
+
+	if (!nrf_rpc_decoding_done_and_check(group, ctx)) {
+		nfc_rpc_report_decoding_error(NFC_RPC_CMD_T2T_CB);
+		return;
+	}
+
+	if (callback) {
+		callback(context, event, data, data_length);
+	}
+
+	nrf_rpc_rsp_send_void(group);
+}
+
+NRF_RPC_CBOR_CMD_DECODER(nfc_group, nfc_rpc_t2t_cb, NFC_RPC_CMD_T2T_CB, nfc_rpc_t2t_cb_rpc_handler,
+			 NULL);
+
+int nfc_t2t_setup(nfc_t2t_callback_t callback, void *context)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = 0;
+	int error;
+
+	cbor_buffer_size += sizeof(uintptr_t) + 1; /* callback */
+	cbor_buffer_size += sizeof(uint32_t) + 1;  /* context */
+
+	NRF_RPC_CBOR_ALLOC(&nfc_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_encode_callback(&ctx, callback);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)context);
+
+	nrf_rpc_cbor_cmd_no_err(&nfc_group, NFC_RPC_CMD_T2T_SETUP, &ctx, nfc_rpc_decode_error,
+				&error);
+
+	return error;
+}
+
+int nfc_t2t_parameter_set(nfc_t2t_param_id_t id, void *data, size_t data_length)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = 1; /* nfc_t2t_param_id_t */
+	int error;
+
+	if (data == NULL) {
+		return -NRF_EINVAL;
+	}
+
+	cbor_buffer_size += data_length + 2;
+
+	NRF_RPC_CBOR_ALLOC(&nfc_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_encode_uint(&ctx, id);
+	nrf_rpc_encode_buffer(&ctx, data, data_length);
+
+	nrf_rpc_cbor_cmd_no_err(&nfc_group, NFC_RPC_CMD_T2T_PARAMETER_SET, &ctx,
+				nfc_rpc_decode_error, &error);
+
+	return error;
+}
+
+int nfc_t2t_parameter_get(nfc_t2t_param_id_t id, void *data, size_t *max_data_length)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	struct nfc_rpc_param_getter getter = {data, max_data_length};
+	size_t cbor_buffer_size = 2 + sizeof(nfc_t2t_param_id_t) + sizeof(size_t);
+
+	if (data == NULL || max_data_length == NULL) {
+		return -NRF_EINVAL;
+	}
+
+	NRF_RPC_CBOR_ALLOC(&nfc_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_encode_uint(&ctx, id);
+	nrf_rpc_encode_uint(&ctx, *max_data_length);
+
+	nrf_rpc_cbor_cmd_no_err(&nfc_group, NFC_RPC_CMD_T2T_PARAMETER_GET, &ctx,
+				nfc_rpc_decode_parameter, &getter);
+
+	return getter.data ? 0 : -NRF_EINVAL;
+}
+
+static int send_payload(enum nfc_rpc_cmd_server cmd, const uint8_t *data, size_t data_length)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size;
+	int error;
+
+	if (data == NULL) {
+		return -NRF_EINVAL;
+	}
+
+	cbor_buffer_size = data_length + 2;
```

Comment:
I think NFC payload is not limited to 255 bytes, right? In such a case, `+ 3` might be a better choice.